### PR TITLE
Added bot system

### PR DESCRIPTION
**Pull request by NEXOmega**
**TODO :** 

- Re-Enable Ressource Pack, disabled because bot would not accept it and stay blocked before joining server
- Finish Command to start test individually

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -36,9 +36,13 @@
     "iterator-helper": "^1.1.3",
     "minecraft-data": "^2.74.0",
     "minecraft-protocol": "^1.23.1",
+    "mineflayer": "^3.7.0",
+    "mineflayer-move": "github:cheezbarger/mineflayer-move",
```
Comment (DeltaEvo) [inline package.json:40]: This should be moved to devDependencies

**Review by DeltaEvo — CHANGES_REQUESTED** [review]
Nice ! Thanks for taking the time to implement that, but I don't think it should be part of the "main" code we should separate the tests from the code,
also I was thinking to use https://jestjs.io/ for the tests

Can you try to separate what you did from the main code (inside a `test` folder instead of `src`) ?

**Comment by Sceat** [comment]
> Nice ! Thanks for taking the time to implement that, but I don't think it should be part of the "main" code we should separate the tests from the code,
> also I was thinking to use [jestjs.io](https://jestjs.io/) for the tests
> 
> Can you try to separate what you did from the main code (inside a `test` folder instead of `src`) ?

We could also use https://github.com/HydreIO/doubt

File: package.json [hunk 2]
```diff
@@ -6,6 +6,7 @@
   "type": "module",
   "scripts": {
     "start": "node src/index.js | pino-human",
+    "test": "node --experimental-vm-modules ./node_modules/jest/bin/jest.js",
```
Comment (Sceat) [inline package.json:9]: Let's discuss before futher implementation as i actually tried Jest and it sucks with esmodules, really don't think we should use it, @hydre/doubt or node-tap are great.

File: src/index.js [hunk 3]
```diff
@@ -143,8 +144,8 @@ const mobs_position = mobs_position_factory(world)
 async function observe_client(context) {
   /* Observers that handle the protocol part.
    * They get the client and should map it to minecraft protocol */
-
-  await player_resource_pack.observe(context)
+  if (!context.client.username.startsWith('#iRobot'))
+    await player_resource_pack.observe(context)
```
Comment (Sceat) [inline src/index.js:148]: probably @DeltaEvo will know how to handle this

File: tests/bot.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import mineflayer from 'mineflayer'
```
Comment (Sceat) [inline tests/bot.js:1]: the folder name should be `test`

File: tests/connected.test.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  return new Promise((resolve) => {
+    bot.on('spawn', () => {
+      resolve(true)
+    })
+  })
+}
+
+test('Can connect', () => {
+  return connected().then((data) => {
+    expect(data).toBeTruthy()
```
Comment (Sceat) [inline tests/connected.test.js:15]: this is great! i like the idea of simply implementing a testing base

File: tests/connected.test.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+import bot from './bot.js'
+
+function connected() {
+  return new Promise((resolve) => {
+    bot.on('spawn', () => {
+      resolve(true)
+    })
+  })
+}
```
Comment (Sceat) [inline tests/connected.test.js:11]:
```suggestion
import { expect, test } from '@jest/globals'
import { once } from 'events'

import bot from './bot.js'

const connected = () => events.once(bot, 'spawn')
```

**Review by Sceat — CHANGES_REQUESTED** [review]
Thanks again 🎉

**Comment by Sceat** [comment]
closing for inactivity

**Pull request closed without merging**